File: Dot examples/ASP Book Examples/Book Examples/Chapter05/CS/App_Code/DynamicMasterPage.cs
using System;
using System.Web.UI;
using System.Web.Profile;

public class DynamicMasterPage : Page
{

    protected override void OnPreInit(EventArgs e)
    {
        this.MasterPageFile = (string)Context.Profile["MasterPageFile"];
        base.OnPreInit(e);
    }

}

File: Dot examples/ASP Book Examples/Book Examples/Chapter05/VB/App_Code/DynamicMasterPage.vb
Imports Microsoft.VisualBasic

Public Class DynamicMasterPage
    Inherits Page

    Protected Overrides Sub OnPreInit(ByVal e As EventArgs)
        Me.MasterPageFile = CType(Context.Profile("MasterPageFile"), String)
        MyBase.OnPreInit(e)
    End Sub

End Class
